packages/block-editor/src/components/block-list/use-block-props/index.js

```diff
@@ -191,7 +191,8 @@ export function useBlockProps( props = {}, { __unstableIsHtml } = {} ) {
 							getBlockName( movingClientId ),
 							getBlockRootClientId( clientId )
 						),
-					[ attributes.className ]: hasLightBlockWrapper,
+					[ attributes.className ]:
+						attributes.className && hasLightBlockWrapper,
 					[ getBlockDefaultClassName( blockName ) ]:
```

Comment:
Could this also be undefined or null? Maybe we shouldn't add either one through an object, and add `hasLightBlockWrapper ? attributes.className : undefined` after it.

Reply:
> Could this also be undefined or null?

This can happen. This is because the `blocks.getBlockDefaultClassName` hook allows consumers to intentionally return `undefined`.

```javascript
wp.hooks.addFilter(
	'blocks.getBlockDefaultClassName',
	'my-plugin/remove-block-class-name',
	function( className, blockName ) {
		return undefined;
	}
);
```

I've updated the PR to take this case into account as well.